CHT/Temperature.C

```diff
@@ -1,59 +1,98 @@
 #include "Temperature.H"
+#include "primitivePatchInterpolation.H"
+
 
 using namespace Foam;
 
 preciceAdapter::CHT::Temperature::Temperature
 (
-    const Foam::fvMesh& mesh,
-    const std::string nameT
-)
-:
-T_(
-    const_cast<volScalarField*>
-    (
-        &mesh.lookupObject<volScalarField>(nameT)
-    )
-)
+        const Foam::fvMesh& mesh,
+        const std::string nameT
+        )
+    :
+      T_(
+          const_cast<volScalarField*>
+          (
+              &mesh.lookupObject<volScalarField>(nameT)
+              )
+          ),
+
+      mesh_(mesh)
+
 {
     dataType_ = scalar;
+
 }
 
-void preciceAdapter::CHT::Temperature::write(double * buffer)
+void preciceAdapter::CHT::Temperature::write(double * buffer, bool provideMeshConnectivity)
 {
+
+    //If we have a nearest projection mapping, we interpolate from the centres to the nodes.
+    //The Node data is then passed to preCICE
+    //TODO: Add the locationsType in the class
```

Comment:
Is this still valid? I am not sure what you mean.

Reply:
Deprecated